src/classes/GiftBatchGroupedTotals.cls

```diff
@@ -0,0 +1,48 @@
+public inherited sharing class GiftBatchGroupedTotals {
+    private final String NPSP_DATA_IMPORT_BATCH =
+        Schema.SObjectType.DataImport__c.fields.NPSP_Data_Import_Batch__c.getName();
+    private Map<Id, GiftBatchTotals> giftBatchTotalsByBatchId = new Map<Id, GiftBatchTotals>();
+
+    public Map<String, Integer> totalsFor(Id batchId) {
+        guardNoBatchTotalsFor(batchId);
+        return giftBatchTotalsByBatchId.get(batchId).toMap();
+    }
+
+    public GiftBatchGroupedTotals(List<AggregateResultProxy> results) {
+        Map<Id, List<AggregateResultProxy>> queryResultsByBatchId = groupQueryResultsByBatchId(results);
+
+        for (Id batchId : queryResultsByBatchId.keySet()) {
+            List<AggregateResultProxy> queryResults = queryResultsByBatchId.get(batchId);
+            giftBatchTotalsByBatchId.put(batchId, new GiftBatchTotals(queryResults));
+        }
+    }
+
+    private Map<Id, List<AggregateResultProxy>> groupQueryResultsByBatchId(List<AggregateResultProxy> results) {
+        Map<Id, List<AggregateResultProxy>> queryResultsByBatchId = new Map<Id, List<AggregateResultProxy>>();
+        for(AggregateResultProxy queryResultProxy : results) {
+            addTo(queryResultsByBatchId, queryResultProxy);
+        }
+        return queryResultsByBatchId;
+    }
+
+    private void addTo(
+        Map<Id, List<AggregateResultProxy>> queryResultsByBatchId,
+        AggregateResultProxy queryResultProxy) {
+
+        Id batchId = (Id) queryResultProxy.getValuesByAlias().get(NPSP_DATA_IMPORT_BATCH);
```

Comment:
`Id batchId = (Id) queryResultProxy.getValuesByAlias().get(String.valueOf
                (DataImport__c.NPSP_Data_Import_Batch__c));`

I'd skip the class variable And just use string.valueof on the field token here.